build.sbt

```diff
@@ -3,11 +3,14 @@ lazy val root = Project(id = "functional-patterns", base = file("."))
   .settings(
     publish         := {},
     publishLocal    := {},
-    publishArtifact := false
+    publishArtifact := false,
+    scalaVersion    := "3.4.0"
```

Comment:
is there any specific Scala 3 feature you felt that could be used in our parser tasks?

Reply:
Not particularly - I should probably do this change in another branch/PR, but did it here at it's a WIP. I'd like to use Scala 3 saying as I have next to no experience in it.

Reply:
https://github.com/monmcguigan/scala-training/pull/7 Have merged the changes in another PR.